plugin.yaml

```diff
@@ -10,58 +10,215 @@ plugins:
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
     interfaces:
       cloudify.interfaces.lifecycle:
         create:
           implementation: gcp_plugin.plugin.tasks.create_network
           inputs:
-            config:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
+            network:
               default: {}
         delete:
           implementation: gcp_plugin.plugin.tasks.delete_network
           inputs:
-            config:
-              default: {}
+            gcp_config:
+              default: { get_property: [SELF, gcp_config]}
 
   cloudify.gcp.nodes.FirewallRule:
     derived_from: cloudify.nodes.Root
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
     interfaces:
       cloudify.interfaces.lifecycle:
         create:
           implementation: gcp_plugin.plugin.tasks.create_firewall_rule
           inputs:
-            config:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config]}
+            firewall_rule:
               default: {}
         delete:
           implementation: gcp_plugin.plugin.tasks.delete_firewall_rule
           inputs:
-            config:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config]}
+            firewall_rule:
               default: {}
 
+  cloudify.gcp.nodes.SecurityGroup:
+    derived_from: cloudify.nodes.SecurityGroup
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      rules:
+        description: >
+          You need to pass in either source_tags list OR cidr_ip list,
+          and then the following two: ip_protocol and port list OR target_tags list.
+        default: []
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create: 
+          implementation: gcp_plugin.plugin.tasks.create_security_group
+          inputs: 
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
+            rules:
+              default: { get_property: [SELF, rules] }
+        delete:          
+          implementation: gcp_plugin.plugin.tasks.delete_firewall_rule
```

Comment:
It's exactly the same function so I want to leave it as it is. If you think this is reasonable I'll make delete_security_group operation calling delete firewall rule.